### DISCUSSION TITLE
Printing data on display

### DISCUSSION THREAD

**Discussion by hugowaals** [Bodmer/TFT_eSPI#1739] [Q&A - General] [answered]
Good day,

I assume that my question is simple, but I don't know the solution: For example, when I want to display a float (**tft.drawFloat(xxx,3,90,140);**) that works fine, but the next print in the same place does not clear the field first. How can I make sure this happens?

Many thanks in advance,
Hugo

**Comment by Bodmer** [comment, marked as answer]
It depends which font you are using. Most fonts allow a background colour to be specified. The setTextPadding() function can also be used to pad out the backgrounbd zone to a specified width in pixels. Some examples demonstrate this.

Example code:
```
  // Set text padding to 100 pixels wide area to over-write old values on screen
  tft.setTextPadding(100);
  tft.setTextColor(TFT_WHITE, TFT_BLUE);
  for (int i = 0; i <= 20; i++) {
    tft.drawFloat(i / 10.0, 1, xpos, ypos);
    delay (200);
  }

```

**Reply by hugowaals** [reply]
Many thanks!!
